osbs/utils/yaml.py

```diff
@@ -62,17 +63,34 @@ def read_yaml(yaml_data, schema, package=None):
     except jsonschema.SchemaError:
         logger.error('invalid schema, cannot validate')
         raise
-    except jsonschema.ValidationError:
+    except jsonschema.ValidationError as exc:
+        logger.debug("schema validation error: %s", exc)
+
+        exc_message = get_error_message(exc)
+
         for error in validator.iter_errors(data):
-            path = "".join(
-                ('[{}]' if isinstance(element, int) else '.{}').format(element)
-                for element in error.path
-            )
+            error_message = get_error_message(error)
 
-            if path.startswith('.'):
-                path = path[1:]
+            logger.debug("validation error: %s", error_message)
 
-            logger.error('validation error (%s): %s', path or 'at top level', error.message)
-        raise
+        raise OsbsValidationException(exc_message)
 
     return data
+
+
+def get_error_message(error):
+    path = "".join(
+        ('[{}]' if isinstance(element, int) else '{}').format(element)
+        for element in error.path
+    )
```

Comment:
The format for a string element should be `.{}`, not just `{}` - otherwise, if an error happens at path `.foo.bar[0].baz`, it would be shown as `foobar[0]baz`. It would be nice to have a test for this, but I'm not sure if the error can happen anywhere other than "at top level" with the container.json schema :thinking:

Reply:
Fixed